grommet/grommet
# Select and SelectMultiple accessibility improved

What does this PR do?
If the Select or SelectMultiple is in a FormField with a label and no placeholder text the label will be read out by a screenreader when the Select or SelectMultiple receives focus.

Since Select and SelectMultiple are complex components htmlFor doesn’t work as intended, so I added a aria-labelledby for Select and SelectMultiple to correctly associate to external label in the FormField.

Where should the reviewer start?
FormField/stories/simple.js

What testing has been done on this PR?
Locally in the scenario where screenreader focuses on Select or SelecMultiple.

How should this be manually tested?
Create a Select component in a FormField with a label and no placeholder text. Then use a screenreader to focus on the Select Component

Do Jest tests follow these best practices?

Any background context you want to provide?

What are the relevant issues?

Screenshots (if appropriate)
Before with incorrect association: 
![Screenshot 2024-10-16 at 4 29 25 PM](https://github.com/user-attachments/assets/caef8bf9-b98b-41be-bd71-ba3b4f3c697f)
 After with correct association:
![Screenshot 2024-10-16 at 4 18 22 PM](https://github.com/user-attachments/assets/4e33d36e-e302-404a-9975-b7691df14b3c)

Do the grommet docs need to be updated?
no

Should this PR be mentioned in the release notes?
no

Is this change backwards compatible or is it a breaking change?
backwards compatible
